engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -908,7 +908,7 @@ private void execute() throws YarnException, IOException
           LOG.info("Releasing {} as resource with priority {} was already assigned", allocatedContainer.getId(), allocatedContainer.getPriority());
           releasedContainers.add(allocatedContainer.getId());
           numReleasedContainers++;
-          numRequestedContainers--;
+          numRequestedContainers++;    // undo the decrement above for this allocated container
```

Comment:
@vrozov  I looked into your suggestion about line 925 (now line 917) and I have convinced myself that no change is necessary there. The reason is as follows:

The current change in this PR (current line 903) relates to the condition of "already allocated container" in which case it makes sense to increment `numRequestedContainers` to undo the bulk decrement that happened on (current) line 882. `numReleasedContainers` ideally needs to track `numRequestedContainers` so that is incremented as well although note that `numReleasedContainers` is not used in any decision making in the code. The "already allocated container" case does not change the fact that so-many containers were requested.

However the other case on line 917 is about "container no longer required" which means the app no longer requires the container originally requested. In that case it makes sense to keep `numRequestedContainers` as decremented and although a request was added to `releasedContainers` there is no need to increment `numReleasedContainers` because it is as though the request and release never took place. In any case `numReleasedContainers` is not used in any decision making.

Regarding testing, let me know how this change can be tested easily - since it involves "already allocated container" which is difficult to simulate.

I have opened APEXCORE-771 for the refactoring suggestion.